fs/fs.go

```diff
@@ -102,7 +104,13 @@ func NewFsInfo(context Context) (FsInfo, error) {
 
 	// need to call this before the log line below printing out the partitions, as this function may
 	// add a "partition" for devicemapper to fsInfo.partitions
-	fsInfo.addDockerImagesLabel(context)
+	if context.DockerRoot != "" {
+		fsInfo.addDockerImagesLabel(context)
+	}
+
+	if context.RktPath != "" {
+		fsInfo.addRktImagesLabel(context)
+	}
```

Comment:
What if instead of hardcoding this, we created an `FsHandler` abstraction, and create a docker & rkt implemention in the respective Container directory, and have each implementation register itself? Then here you just loop over all the registered `FsHandler`s and call `handler.AddImagesLabel`?